frontend/__tests__/src/hooks/useMetricsStepValidationCheckContext.test.tsx

```diff
@@ -0,0 +1,279 @@
+import React from 'react'
+import { addADeploymentFrequencySetting, updateDeploymentFrequencySettings } from '@src/context/Metrics/metricsSlice'
+import { Provider } from 'react-redux'
+import { setupStore } from '../utils/setupStoreUtil'
+import { ToolkitStore } from '@reduxjs/toolkit/dist/configureStore'
+
+describe('useMetricsStepValidationCheckContext', () => {
+  const duplicatedData = [
+    {
+      updateId: 0,
+      label: 'organization',
+      value: 'mockOrganization',
+    },
+    { updateId: 0, label: 'pipelineName', value: 'mockPipelineName' },
+    { updateId: 0, label: 'steps', value: 'mockSteps' },
+    { updateId: 1, label: 'organization', value: 'mockOrganization' },
+    { updateId: 1, label: 'pipelineName', value: 'mockPipelineName' },
+    { updateId: 1, label: 'steps', value: 'mockSteps' },
+  ]
+
+  const duplicatedDataErrorMessages = [
+    {
+      id: 0,
+      error: {
+        organization: 'duplicated organization',
+        pipelineName: 'duplicated pipelineName',
+        steps: 'duplicated steps',
+      },
+    },
+    {
+      id: 1,
+      error: {
+        organization: 'duplicated organization',
+        pipelineName: 'duplicated pipelineName',
+        steps: 'duplicated steps',
+      },
+    },
+  ]
+
+  const setup = () => {
+    const store = setupStore()
+    const wrapper = ({ children }: { children: React.ReactNode }) => (
+      <Provider store={store}>
+        <ContextProvider>{children}</ContextProvider>
+      </Provider>
+    )
+    const { result } = renderHook(() => useMetricsStepValidationCheckContext(), { wrapper })
+
+    return { result, store }
+  }
+
+  const setDuplicatedDataToStore = (store: ToolkitStore) => {
+    store.dispatch(addADeploymentFrequencySetting())
+    duplicatedData.map((data) => {
+      store.dispatch(updateDeploymentFrequencySettings(data))
+    })
+  }
+
+  it('should return initial ValidationContext ', () => {
+    const { result } = renderHook(() => useMetricsStepValidationCheckContext())
+
+    expect(result.current?.errorMessages).toEqual([])
+    expect(result.current?.clearErrorMessage(1, 'label')).toBe(null)
+    expect(result.current?.checkDuplicatedPipeLine()).toBe(null)
+    expect(result.current?.isPipelineValid()).toBe(false)
+  })
+
+  it('should return useMetricsStepValidationCheckContext correctly ', () => {
+    const { result } = setup()
+
+    expect(result.current?.errorMessages).toEqual([])
+    expect(result.current?.clearErrorMessage).toBeInstanceOf(Function)
+    expect(result.current?.checkDuplicatedPipeLine).toBeInstanceOf(Function)
+    expect(result.current?.isPipelineValid).toBeInstanceOf(Function)
+  })
+
+  it('should clear error message when call clearErrorMessage given error message', async () => {
+    const { result } = setup()
+
+    act(() => {
+      result.current?.isPipelineValid()
+    })
+
+    await waitFor(() => {
```

Comment:
Will an error be reported when deleting the await waitFor code? It seems like using act, but it's okay not to wait for